teps/draft-20210126-pipeline-task-display-name.md

```diff
@@ -0,0 +1,119 @@
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+A task in a Pipeline is currently represented in the UI using a field (`name`) that is meant to be
+machine readable, not human readable. There should be the addition of a new
+optional field (`displayName`) to describe Pipeline tasks that are human readable.
+
+## Motivation
+
+Simple version, humans want to read task names formatted for human's not computers, and they want
+to see the full task name rendered in their UI.
+
+
+The longer version, the end user of a Tekton pipeline can vary including but not limited to
+application developer, to security professional, to compliance officer,
+service reliability engineer, product manager, and beyond. The farther
+you move away from the application developer the more important human
+understanding of the CI/CD process is important.
+
+Currently if a product manager or auditor was to look at a Tekton Pipeline run rendered in UI
+that would be presented with a bunch of truncated machine readable task names, no spaces, no
+capitalization, and the entire task name isn't even readable beyond 20 or so characters.
+
+Compare this to other tools in this space, Jenkins, GitLab CI, GitHub Actions, etc.
+Their renderings of their workflows are meant to be human consumable. Their task names have
+the option to be rendered in a human consumable format with use of any characters and fully
+rendered for context.
+
+### Goals
+
+* Add a `Pipeline.tasks.*.displayName` (or similar) field that allows any text and
+  if provided will be rendered when displaying Tekton Pipeline in UI rather then the
+  machine readable `Pipeline.tasks.*.name` field.
+* Ideally when rendering this new display name it will be fully rendered rather then the
+  current truncation to 20 characters that is currently happening.
+
+### Non-Goals
+
+To be determined.
+
+### Use Cases (optional)
+
+Pipeline writers can specify a display name for any/all tasks, if supplied that display name
+will be rendered in UI rather then the machine readable `name` field value.
+
+## Requirements
+
+New display name filed would be optional and accept any unicode character
+
+## Proposal
+
+Add `Pipeline.tasks.*.displayName` and update Tekton rendering to use that field rather then
+`Pipeline.tasks.*.name` when provided.
```

Comment:
@itewk you proposing to add a displayName under `pipelineTasks`?

```
kind: PipelineRun
spec:
  pipelineSpec:
    tasks:
      - name: echo-message
         displayName: Echo Message
         taskSpec:
           steps:
             - name: echo
               image: ubuntu
               script: |
                 echo "Hi!"
```

Reply:
@pritidesai yes.

Reply:
thanks @itewk please include the sample usage in the TEP if possible saying its `optional`.

Reply:
Also, What is the expectation in terms of validation of such names? 

`pipelineTask` names are validated to make sure they comply with RFC 1123.

https://github.com/tektoncd/pipeline/blob/d4a3aa2ec86e4711490d24fa185502da794ef426/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L91-L100

Should follow the same validation process?

Reply:
I would believe displayName should not have a strong validation and let the user set it to whatever we want?

Reply:
deffinitly not the RFC 1123, as its not meant to be read by computer, and just for display, it should be free form and allow full set of unicode chars so people can do display names with none roman characters if so desire

Reply:
example pushed.